### PR TITLE
[Main Course] - 使用TensorFlow进行文本分类

### PR DESCRIPTION
**Pull request by Realvincentyuan**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (使用TensorFlow进行文本分类)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Deep_Learning/2022-04-16-TF-Text-Classification.md [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+
+for pattern, rewrite in str_regex_pattern:
+  train_ds = train_ds.map(lambda text, label: (tf.strings.regex_replace(text, pattern=pattern, rewrite=rewrite), label))
+  val_ds = val_ds.map(lambda text, label: (tf.strings.regex_replace(text, pattern=pattern, rewrite=rewrite), label))
+
+```
+
+### 2.3 构建词表
+
+使用训练集构造词表（注意不要使用验证集或者测试集，会导致信息泄露），改步骤将字符映射到相应的索引，利于将数据转化为模型能够进行训练和预测的格式。
```
Comment (jhqian0909) [inline docs/Main_Course/Deep_Learning/2022-04-16-TF-Text-Classification.md:123]: 该步骤

**Review by jhqian0909 — APPROVED** [review]
LGTM 感谢分享

File: docs/Main_Course/Deep_Learning/2022-04-16-TF-Text-Classification.md [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+<class 'tensorflow.python.framework.ops.EagerTensor'>
+Text: b"This was an absolutely terrible movie. Don't be lured in by Christopher Walken or Michael Ironside. Both are great actors, but this must simply be their worst role in history. Even their great acting could not redeem this movie's ridiculous storyline. This movie is an early nineties US propaganda piece. The most pathetic scenes were those when the Columbian rebels were making their cases for revolutions. Maria Conchita Alonso appeared phony, and her pseudo-love affair with Walken was nothing but a pathetic emotional plug in a movie that was devoid of any real meaning. I am disappointed that there are movies like this, ruining actor's like Christopher Walken's good name. I could barely sit through it."
+Label: 0
+```
+
+## 2 文本预处理
+
+该小节使用tf_text和tf.stings的处理文本的API对数据进行处理，tf.data.Dataset能够很方便得将对应的函数映射到数据中，推荐学习和使用。
+
+### 2.1 讲文字转为小写
```
Comment (EchoTang04) [inline docs/Main_Course/Deep_Learning/2022-04-16-TF-Text-Classification.md:98]: 将文字

File: docs/Main_Course/Deep_Learning/2022-04-16-TF-Text-Classification.md [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+  test_ds = test_ds.map(lambda text, label: (tf.strings.regex_replace(text, pattern=pattern, rewrite=rewrite), label))
+
+test_ds = test_ds.map(text_index_lookup)
+test_ds = configure_dataset(test_ds)
+test_ds = test_ds.padded_batch(BATCH_SIZE  )
+```
+
+## 3 建立模型
+
+## 使用Sequential API构建卷积神经网络
```
Comment (EchoTang04) [inline docs/Main_Course/Deep_Learning/2022-04-16-TF-Text-Classification.md:269]: 这里是##还是###？是跟上面‘3.建立模型’并列还是是它的一个分点呀？

File: docs/Main_Course/Deep_Learning/2022-04-16-TF-Text-Classification.md [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+
+考虑到模型结构简单，效果还可以接受：
+
+```
+782/782 [==============================] - 57s 72ms/step - loss: 0.4583 - accuracy: 0.8678
+Loss:  0.45827823877334595
+Accuracy: 86.78%
+```
+
+## 使用Functional API构建双向LSTM
```
Comment (EchoTang04) [inline docs/Main_Course/Deep_Learning/2022-04-16-TF-Text-Classification.md:362]: 同上

**Review by EchoTang04 — APPROVED** [review]
LGTM, 感谢袁老板分享！

**Review by Tina0330 — APPROVED** [review]
感谢袁老板分享！

File: docs/Main_Course/Deep_Learning/2022-04-16-TF-Text-Classification.md (hunk 3 above)
Comment (Realvincentyuan) [reply docs/Main_Course/Deep_Learning/2022-04-16-TF-Text-Classification.md]: 已经修复

**Pull request merged**